pkg/appclient/endpoint_selector.go

```diff
@@ -0,0 +1,38 @@
+package appclient
+
+import (
+	"context"
+	"net/url"
+
+	sessiontypes "pocket/x/session/types"
+	sharedtypes "pocket/x/shared/types"
+)
+
+// getRelayerUrl gets the URL of the relayer for the given service.
+// It gets the suppliers list from the current session and returns
+// the first relayer URL that matches the JSON RPC RpcType.
```

Comment:
Why not make `JSON_RPC` a parameter and make this auto-customizable right away?

Otherwise, it should probably be part of the `jsonrpc` file.